packages/devextreme/js/__internal/core/state_manager/observable_state_container_manager.ts

```diff
@@ -0,0 +1,189 @@
+import { Observable } from '../reactive/core';
+import type {
+  ILogger,
+  IMaybeObservableStateContainer,
+  IObservableStateContainer,
+  IStateChange,
+  IStateChangeCallback,
+  IStateChangePayload,
+  IStateContainerManager,
+} from './interfaces';
+import { areEqual, deepCopy } from './utils';
+
+export class ObservableStateContainerManager implements IStateContainerManager {
+  private readonly logger: ILogger;
+
+  private readonly controllerSign: string;
+
+  constructor(logger: ILogger, controllerSign: string) {
+    this.logger = logger;
+    this.controllerSign = controllerSign;
+  }
+
+  canHandle(
+    stateContainer: IMaybeObservableStateContainer,
+  ): stateContainer is IObservableStateContainer {
+    return stateContainer instanceof Observable;
+  }
+
+  trackChanges(
+    stateContainer: IMaybeObservableStateContainer,
+    stateName: string,
+    onChange: IStateChangeCallback,
+  ): void {
+    if (!this.canHandle(stateContainer)) {
+      this.logger.error('State container is not an Observable');
+      return;
+    }
+
+    if (!stateName) {
+      this.logger.error('State name is required');
+      return;
+    }
+
+    if (!onChange || typeof onChange !== 'function') {
+      this.logger.error('onChange callback is required');
+      return;
+    }
+
+    // eslint-disable-next-line spellcheck/spell-checker
+    let previousValue = stateContainer.unreactive_get();
+
+    const originalUpdate = stateContainer.update;
```

Comment:
I just have to let it out: I'm still pessimistic about "a hive of atoms" - decentralized state with individual state manupulations. Though I must admit, it looks interesting.

Anyway it's out of this PR scope, I'm just sharing my thoughts.